packages/headless/src/features/facets/facet-set/facet-set-slice.ts

```diff
@@ -189,27 +189,29 @@ export const facetSetReducer = createReducer(
         }
 
         const searchResultValue = buildSelectedFacetValueRequest(rawValue);
-        const firstIdleIndex = currentValues.findIndex(
-          (v) => v.state === 'idle'
-        );
-        const indexToInsertAt =
-          firstIdleIndex === -1 ? currentValues.length : firstIdleIndex;
-
-        const valuesBefore = currentValues.slice(0, indexToInsertAt);
-        const valuesAfter = currentValues.slice(indexToInsertAt + 1);
-
-        facetRequest.currentValues = [
-          ...valuesBefore,
-          searchResultValue,
-          ...valuesAfter,
-        ];
-        facetRequest.numberOfValues = facetRequest.currentValues.length;
+        insertValueBeforeIdleValues(facetRequest, searchResultValue);
         facetRequest.freezeCurrentValues = true;
         facetRequest.preventAutoSelect = true;
       });
   }
 );
 
+function insertValueBeforeIdleValues(
```

Comment:
The `before` in this function name is slightly misleading. The insert happens "in place" of the first idle value, or at the end of the array if there are no idle values.

Suggestion: simplify the name  to `insertValue` or `addNewValue` etc. I don't think explaining in the name the details of how the insertion is happening is important.